ogr/services/gitlab/project.py

```diff
@@ -210,21 +209,15 @@ def _get_collaborators_with_given_access(
             if member.access_level in access_levels
         ]
 
-    def get_issue_comments(
-        self, issue_id, filter_regex: str = None, reverse: bool = False
-    ) -> List["IssueComment"]:
+    def _get_all_issue_comments(self, issue_id) -> List["IssueComment"]:
         issue = self.gitlab_repo.issues.get(issue_id)
-        if not filter_regex:
-            filter_regex = r".*"
 
         comments = [
             self._issuecomment_from_gitlab_object(raw_comment)
             for raw_comment in issue.notes.list()
-            if re.search(filter_regex, raw_comment.body)
         ]
 
-        if not reverse:
-            comments.reverse()
+        comments.reverse()
```

Comment:
What about `return reversed(comments)` so we postpone the reversing?

Ideally, we can check the type for `list_reverseiterator` in the `__get_comments` to not reverse twice. (But I am not sure if we can get the original list from the `reversed` so its probably a bad idea. We will need to use custom version of reversed to make it work..;-)

But please, do it only if the code stays readable..;-)

I hope you know `this`:

```python
import this
```                                                          
The Zen of Python, by Tim Peters

***Beautiful is better than ugly.***
Explicit is better than implicit.
***Simple is better than complex.***
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
***Readability counts.***
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
***There should be one-- and preferably only one --obvious way to do it.***
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!

Reply:
I'll have a look if it's possible to get it in right order from the library... Only the gitlab returns newest first... I've also changed it for PRs, it was returning newest first too, so it's same for all services right now...

Checking the type would be also used in github and pagure implementation then, where it's not needed

Reply:
OK, if you think the current approach is good enough, ping me and we can merge.

Reply:
I knew I've seen it in the API... They just don't have it in the documentation of the `python-gitlab`... No reverse needed anymore